src/k2/httpproxy/HTTPProxy.cpp

```diff
@@ -209,12 +218,12 @@ seastar::future<nlohmann::json> HTTPProxy::_handleEnd(nlohmann::json&& request)
         return JsonResponse(Statuses::S400_Bad_Request("Bad json for end request"));
     }
 
-    std::unordered_map<uint64_t, k2::K2TxnHandle>::iterator it = _txns.find(id);
-    if (it == _txns.end()) {
+    auto iter = _txns.find(id, false);
+    if (iter == _txns.end()) {
         return JsonResponse(Statuses::S400_Bad_Request("Could not find txnID for end request"));
     }
 
-    return it->second.end(commit)
+    return _txns.getValue(iter).txn.end(commit)
     .then([this, id] (k2::EndResult&& result) {
```

Comment:
The queries need to be cleaned up here too right?

Reply:
When TxnInfo is deallocated, queries will be destroyed as query hash table is inside txn now.